geospaas_harvesting/ingesters.py

```diff
@@ -510,6 +513,67 @@ def _get_normalized_attributes(self, dataset_info, *args, **kwargs):
         return normalized_attributes
 
 
+class NetCDFIngester(MetanormIngester):
+    """Ingests metadata from NetCDF files. The files can be either
+    local or remote (if the remote repository supports it).
+    """
+
+    def _get_geometry_wkt(self, dataset):
+        """Extracts the WKT string for the dataset's spatial coverage
+        """
+        raise NotImplementedError()
+
+    def _get_raw_attributes(self, dataset_path):
+        """Get the raw metadata from the NetCDF file"""
+        dataset = netCDF4.Dataset(dataset_path)
+        raw_attributes = dataset.__dict__
+        self.add_url(dataset_path, raw_attributes)
+        raw_attributes['geometry'] = self._get_geometry_wkt(dataset)
+        raw_attributes['raw_dataset_parameters'] = self._get_parameter_names(dataset)
+        return raw_attributes
+
+    def _get_parameter_names(self, dataset):
+        """Get the names of the dataset's variables"""
+        return [
+            variable.standard_name
+            for variable in dataset.variables.values()
+            if hasattr(variable, 'standard_name')
```

Comment:
how about the cases that the writer person of netcdf file does not provide a standard name for the variable? Then we don't add that?

Reply:
No we don't. If they don't have a standard_name, they are probably just metadata variables (for example, in CMEMS in situ data, there are quality control variables which indicate the trustworthiness of "real variables").

Reply:
one another aspect is that:
How are you certain that the `variable.standard_name` is really written in standard way? I mean isn't it necessary to normalize it?

Reply:
It is still necessary to normalize it, and it is done by calling metanorm like in other ingesters.